### ISSUE TITLE
Invocation of `pulumi state move` panics with `attempt to encrypt value`

### ISSUE THREAD

**Issue by koozz** [pulumi/pulumi#16874]
### What happened?

When splitting up a big stack into independent smaller stacks, the Pulumi CLI panics.

Quite a few resource types succeeded, but a few keep failing (repeatedly).

They seem to be related to an OpsGenie resource (User/Team/NotificationPolicy), but this might also be a coincidence.

### Example

```shell
$ pulumi state move --source teams --dest demo <urn to an opsgenie:index/user:User> --yes

================================================================================
The Pulumi CLI encountered a fatal error. This is a bug!
We would appreciate a report: https://github.com/pulumi/pulumi/issues/
Please provide all of the below text in your report.
================================================================================
Pulumi Version:   v3.127.0
Go Version:       go1.22.5
Go Compiler:      gc
Architecture:     arm64
Operating System: darwin
Panic:            attempt to encrypt value

goroutine 1 [running]:
runtime/debug.Stack()
        /opt/homebrew/Cellar/go/1.22.5/libexec/src/runtime/debug/stack.go:24 +0x64
main.panicHandler(0x14001b5df1f)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/cmd/pulumi/main.go:34 +0x44
panic({0x103b01b20?, 0x104091d10?})
        /opt/homebrew/Cellar/go/1.22.5/libexec/src/runtime/panic.go:770 +0x124
github.com/pulumi/pulumi/sdk/v3/go/common/resource/config.panicCrypter.EncryptValue(...)
        /private/tmp/pulumi-20240725-4860-tou63g/sdk/go/common/resource/config/crypt.go:103
github.com/pulumi/pulumi/pkg/v3/resource/stack.SerializePropertyValue({0x1040c0d28, 0x1057d8860}, {{0x103a84d40?, 0x14002561460?}}, {0x1063b78d0, 0x1057d8860}, 0x0)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/resource/stack/deployment.go:478 +0x55c
github.com/pulumi/pulumi/pkg/v3/resource/stack.SerializeProperties({0x1040c0d28, 0x1057d8860}, 0x14002a98e40, {0x1063b78d0, 0x1057d8860}, 0x0)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/resource/stack/deployment.go:386 +0xa8
github.com/pulumi/pulumi/pkg/v3/resource/stack.SerializeResource({_, _}, _, {_, _}, _)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/resource/stack/deployment.go:317 +0x178
github.com/pulumi/pulumi/pkg/v3/resource/stack.SerializeDeployment({0x1040c0d28, 0x1057d8860}, 0x140021018c0, 0x0)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/resource/stack/deployment.go:125 +0x288
main.saveSnapshot({0x1040c0d28, 0x1057d8860}, {0x1040db778, 0x14002114528}, 0x140021018c0, 0x0)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/cmd/pulumi/stack_import.go:150 +0x488
main.(*stateMoveCmd).Run(0x14001ef6600, {0x1040c0d28, 0x1057d8860}, {0x1040db778, 0x14002114060}, {0x1040db778, 0x14002114528}, {0x14001f9d680, 0x1, 0x6}, ...)
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/cmd/pulumi/state_move.go:370 +0x1d30
main.newStateMoveCommand.func1(0x0?, {0x14001f9d680, 0x1, 0x6})
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/cmd/pulumi/state_move.go:89 +0x1dc
main.newStateMoveCommand.RunFunc.func2(0x0?, {0x14001f9d680?, 0x0?, 0x0?})
        /private/tmp/pulumi-20240725-4860-tou63g/sdk/go/common/util/cmdutil/exit.go:99 +0x28
main.newStateMoveCommand.RunFunc.RunResultFunc.func5(0x14001fa0308, {0x14001f9d680, 0x1, 0x6})
        /private/tmp/pulumi-20240725-4860-tou63g/sdk/go/common/util/cmdutil/exit.go:115 +0x34
github.com/spf13/cobra.(*Command).execute(0x14001fa0308, {0x14001f9d620, 0x6, 0x6})
        /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:987 +0x828
github.com/spf13/cobra.(*Command).ExecuteC(0x14001027b08)
        /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x344
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.main()
        /private/tmp/pulumi-20240725-4860-tou63g/pkg/cmd/pulumi/main.go:55 +0x54
```

### Output of `pulumi about`

```
CLI
Version      3.127.0
Go Version   go1.22.5
Go Compiler  gc

Plugins
KIND      NAME    VERSION
language  nodejs  unknown

Host
OS       darwin
Version  14.6
Arch     arm64

This project is written in nodejs: executable='/opt/homebrew/bin/node' version='v22.4.1'

Current Stack: organization/teams/demo

TYPE                                                    URN
pulumi:pulumi:Stack                                     urn:pulumi:demo::*redacted*
pulumi:providers:azuread                                urn:pulumi:demo::teams::pulumi:providers:azuread::default_5_53_2
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
pulumi:providers:github                                 urn:pulumi:demo::teams::pulumi:providers:github::default_5_26_0
github:index/team:Team                                  urn:pulumi:demo::*redacted*
github:index/team:Team                                  urn:pulumi:demo::*redacted*
github:index/team:Team                                  urn:pulumi:demo::*redacted*
github:index/teamSyncGroupMapping:TeamSyncGroupMapping  urn:pulumi:demo::*redacted*
github:index/team:Team                                  urn:pulumi:demo::*redacted*
github:index/teamSyncGroupMapping:TeamSyncGroupMapping  urn:pulumi:demo::*redacted*
github:index/team:Team                                  urn:pulumi:demo::*redacted*
github:index/teamSyncGroupMapping:TeamSyncGroupMapping  urn:pulumi:demo::*redacted*
github:index/team:Team                                  urn:pulumi:demo::*redacted*
azuread:index/group:Group                               urn:pulumi:demo::*redacted*
github:index/teamSyncGroupMapping:TeamSyncGroupMapping  urn:pulumi:demo::*redacted*
azuread:index/groupMember:GroupMember                   urn:pulumi:demo::*redacted*
azuread:index/groupMember:GroupMember                   urn:pulumi:demo::*redacted*


Found no pending operations associated with demo

Backend
Name           *redacted*
URL            azblob://pulumi?storage_account=*redacted*
User           jan
Organizations
Token type     personal

Pulumi locates its logs in /var/folders/ws/03wrvzj554j9rsgbsljj2dgh0000gn/T/ by default
```

### Additional context

Verbose logging doesn't show anything extra, seems like there is no verbose logging being triggered.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by tgummerer** [comment]
Thanks for this report.

I'm going to close this out as a duplicate of https://github.com/pulumi/pulumi/issues/16835.  (Note that that issue is closed already, but the fix is not released quite yet (it will be released in the 3.128.0 release, which should be out today)).

In the meantime the workaround in https://github.com/pulumi/pulumi/issues/16835#issuecomment-2262991167, should get you unblocked (you can use `pulumi stack --stack <stack-name> change-secrets-provider default`, to get a secrets manager set up in the destination stack).

This is also the reason why some resources succeed, and others not.  Presumably the resources that succeed being moved don't contain any secrets, while the resources that do, do contain secrets.

**Comment by koozz** [comment]
Thanks @tgummerer, I quickly searched for the panic title (attempt to encrypt value) and didn't find anything. Will attempt some more searches on any next bug report!

**Comment by tgummerer** [comment]
No worries, I'd rather have a duplicate bug report than none at all!
